build/vsts_build.yaml

```diff
@@ -345,14 +345,14 @@ phases:
     inputs:
       command: 'custom'
       arguments: 'sources add -Name VS -Source $(VsPackageFeedUrl) -UserName $(VsPackageFeedUsername) -Password $(VsPackageFeedPassword) -ConfigFile $(System.DefaultWorkingDirectory)\NuGet.config'
-    condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'), eq(variables['ShouldSkipOptimize'], 'false'))"
+    condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'), eq(variables['IsOfficialBuild'], 'true'))"
```

Comment:
Why are we removing these? 

Aren't they're useful for disabling again if we have to?

Reply:
Why should they always run?

Reply:
The clause should not have been added in the [first place](https://github.com/NuGet/NuGet.Client/pull/3057).

These OptProfV2 tasks are not always run.  They are only run on release branches if the build is successful so far.  Task execution should not be conditioned on the value of `ShouldSkipOptimize`.  The purpose of `ShouldSkipOptimize` is to temporarily disable optimization (when `ShouldSkipOptimize` is `true`) so as to unblock the build and allow new optimization profiling data to be created through these OptProfV2 tasks.  Then, `ShouldSkipOptimize` should be reverted to the default value of `false`.

As long as optimization profiling data exists, there should be no need to toggle `ShouldSkipOptimize`.

Reply:
Agree with Damon. The idea is to always generate training data, even when we're not creating optimized binaries (when ShouldSkipOptimize = true)